src/charm.py

```diff
@@ -207,23 +193,32 @@ def _on_collect_unit_status(self, event: ops.CollectStatusEvent) -> None:
         """Handle collect unit status event (called after every event)."""
         if not self.model.relations.get("credentials"):
             event.add_status(BlockedStatus("Keystone is not related"))
+            return
```

Comment:
Are you sure you want to `return`? The status works like a "priority queue", you can add multiple statuses in one go, and the it will be evaluated and prioritize based on the importance mentioned [here](https://ops.readthedocs.io/en/latest/#ops.CollectStatusEvent). The final status is set based on the "priority queue" so you don't need to **explicitly** order the status in this handler.

Reply:
I didn't know about this. Thanks for the information